hub/policy/src/main/java/uk/gov/ida/hub/policy/PolicyModule.java

```diff
@@ -92,9 +93,8 @@ protected void configure() {
     }
 
     @Provides
-    @Singleton
-    public EventEmitter getEventEmitter(PolicyConfiguration configuration) {
-        return new EventEmitter(configuration.getSendToRecordingSystem());
+    private Optional<Configuration> getEventEmitterConfiguration(final PolicyConfiguration configuration) {
+        return Optional.ofNullable(configuration.getEventEmitterConfiguration());
```

Comment:
Why does this need to be an optional?

Reply:
Policy, SAML SOAP Proxy, SAML Proxy configuration files have no EventEmitterConfiguration. The EventEmitterConfiguration will be null and Guice does not like having null value. That's why we use Optional.

Reply:
How come they can need the EventEmitter without having a configuration - do they not need an 
 `EventEmitterConfiguration` in order to get an `EventEmitter`?

Reply:
If Event Emitter Configuration containing a source queue name is not provided, Event Emitter will use stub queue by default. Stub queue does not need source queue name and it writes the encrypted event message to a standard output.